components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -599,7 +608,7 @@ ::update_spa_data_from_file(
   Kokkos::deep_copy(target_grid_loc_h, spa_horiz_interp.target_grid_loc);
   for (int idx=0;idx<spa_horiz_interp.length;idx++) {
     auto src_wgt = weights_h(idx);
-    int  src_col = source_grid_loc_h(idx);
+    int  src_col = spa_horiz_interp.source_local_col_map[source_grid_loc_h(idx)];
```

Comment:
Something seems overly complicated here. Do we need to store the src grid gids, if all you need is their local id? Instead of the map, you could store a vector of LIDs (of size `spa_horiz_interp.length`), specifying the LID of the current src col in the unique grid dof list:
```
int src_col = spa_horiz_interp.source_local_col_lid[idx];
```